Pal/Console.Uefi.cs

```diff
@@ -0,0 +1,143 @@
+﻿namespace System
+{
+    public static class Console
+    {
+        static ushort s_consoleAttribute;
+        static ushort s_cursorX;
+        static ushort s_cursorY;
+        static ushort s_windowSizeX;
+        static ushort s_windowSizeY;
+
+        public static unsafe string Title
+        {
+            set
+            {
+            }
+        }
+
+        public static unsafe bool CursorVisible
+        {
+            set
+            {
+                EfiRuntimeHost.SystemTable->ConOut->EnableCursor(EfiRuntimeHost.SystemTable->ConOut, value);
+            }
+        }
+
+        public unsafe static ConsoleColor ForegroundColor
+        {
+            set
+            {
+                s_consoleAttribute = (ushort)value;
+                uint color = s_consoleAttribute;
+                EfiRuntimeHost.SystemTable->ConOut->SetAttribute(EfiRuntimeHost.SystemTable->ConOut, color);
+            }
+        }
+
+
+        static char lastKey = '\0';
+        static ushort lastScanCode;
+        public static unsafe bool KeyAvailable
+        {
+            get
+            {
+                if (lastKey == '\0')
+                {
+                    return true;
+                }
+
+                EFI_INPUT_KEY key;
+                var errorCode = EfiRuntimeHost.SystemTable->ConIn->ReadKeyStroke(EfiRuntimeHost.SystemTable->ConIn, &key);
+                lastKey = (char)key.UnicodeChar;
+                return errorCode == 0;
+            }
+        }
+
+        public static unsafe void Clear()
+        {
+            EfiRuntimeHost.SystemTable->ConOut->ClearScreen(EfiRuntimeHost.SystemTable->ConOut);
+        }
+
+        public static unsafe ConsoleKeyInfo ReadKey(bool intercept)
+        {
+            if (lastKey == '\0')
+            {
+                EFI_INPUT_KEY key;
+                var errorCode = EfiRuntimeHost.SystemTable->ConIn->ReadKeyStroke(EfiRuntimeHost.SystemTable->ConIn, &key);
+                lastKey = (char)key.UnicodeChar;
+                lastScanCode = key.ScanCode;
+            }
+
+            char c = lastKey;
+
+            // Interpret WASD as arrow keys.
+            ConsoleKey k = default;
+            if (c == 'w')
+                k = ConsoleKey.UpArrow;
+            else if (c == 'd')
+                k = ConsoleKey.RightArrow;
+            else if (c == 's')
+                k = ConsoleKey.DownArrow;
+            else if (c == 'a')
+                k = ConsoleKey.LeftArrow;
```

Comment:
I have this in the DOS version because I couldn't quickly find a DOS API to read arrow keys. But if arrow keys work through the scancodes below, we don't need to carry over this DOS hack.